pkg/manifests/manifests.go

```diff
@@ -463,6 +464,34 @@ func (f *Factory) AlertmanagerMain(host string, trustedCABundleCM *v1.ConfigMap)
 		a.Spec.Storage = &monv1.StorageSpec{
 			VolumeClaimTemplate: *f.config.ClusterMonitoringConfiguration.AlertmanagerMainConfig.VolumeClaimTemplate,
 		}
+	} else if f.infrastructure.HighlyAvailableInfrastructure() {
+		// When no persistent storage is configured, add a startup probe to
+		// ensure that the Alertmanager container has time to replicate data
+		// from other peers before declaring itself as ready. This allows
+		// silences and notifications to be preserved on roll-outs. We assume
+		// that 20 seconds is enough for a full synchronization (this is twice
+		// the time Alertmanager waits before declaring that it can start
+		// sending notfications).
+		a.Spec.Containers = append(a.Spec.Containers,
```

Comment:
Should this change be made in the prometheus operator?

Reply:
Yes it makes sense to follow up upstream. BTW I notice that we have no readiness probe for Alertmanager...